### PR TITLE
WFCORE-1469 - Support match-comparison operation for if-command

### PR DESCRIPTION
**Pull request by thomasdarimont**
Introduced regex based match comparison-operation for if-commands.
The match operation is denoted by the symbol `~=`.
The left operand is a string to match against the right operand
which is a regex string.

This enables a simple way to do feature flag detection
via partial regex matches.

E.g. given the system property "features" which contains a space
separated list of enabled "features".
```
-Dfeatures="activemq jgroups"
```

One could conditionally configure wildfly components via CLI
in one generic CLI script, e.g.:
```
if (features ~= ".*activemq.*") of /:resolve-expression(expression=${features})
echo configuring activemq
end-if

if (features ~= ".*jgroups.*") of /:resolve-expression(expression=${features})
echo configuring jgroups
end-if

if (features ~= ".*postgres.*") of /:resolve-expression(expression=${features})
echo configuring postgres
end-if
```

### CODE REVIEW

**Comment by aloubyansky** [comment]
Looks good to me. Thanks! Could you please also add a simple test to the testsuite too? Something like org.jboss.as.test.integration.management.cli.ifelse.BasicIfElseTestCase? Just have it there in action. Thanks again!

**Comment by thomasdarimont** [comment]
I ammended the commit by adding the mentioned test case.

File: testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/ifelse/BasicIfElseTestCase.java [hunk 1]
```diff
@@ -51,14 +51,48 @@ public void testMain() throws Exception {
...
+            ctx.terminateSession();
+            cliOut.reset();
+        }
+    }
+
     protected String runIf(CommandContext ctx) throws Exception {
         ctx.handle("if result.value==\"true\" of " + this.getReadPropertyReq());
-        ctx.handle(this.getWritePropertyReq("\"false\""));
-        ctx.handle("else");
         ctx.handle(this.getWritePropertyReq("\"true\""));
```
Comment (aloubyansky) [inline testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/ifelse/BasicIfElseTestCase.java]: What is the reason behind this change?

**Comment by thomasdarimont** [comment]
Updated the PR again since I had edits in an unrelated test case.

**Comment by aloubyansky** [comment]
Thanks a lot! This commit looks good to me.

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Thanks for the patch, before it can be merged (assuming you are not a Red Hat employee), we need to do a bit of paperwork.

We require that all contributions be made under the terms of the MIT License, http://www.opensource.org/licenses/mit-license.php. Do you agree to the these terms?

**Comment by thomasdarimont** [comment]
Hello @kabir 

yes, I agree to these terms.

Cheers,
Thomas
